applications/GeoMechanicsApplication/custom_strategies/strategies/geo_mechanics_newton_raphson_erosion_process_strategy.hpp

```diff
@@ -306,25 +304,22 @@ class GeoMechanicsNewtonRaphsonErosionProcessStrategy
         return max_pipe_height / (n_steps - 1);
     }
 
-    bool Recalculate()
+    virtual bool Recalculate()
     {
         KRATOS_INFO_IF("ResidualBasedNewtonRaphsonStrategy", this->GetEchoLevel() > 0 && rank == 0)
             << "Recalculating" << std::endl;
-        // KRATOS_INFO_IF("PipingLoop") << "Recalculating" << std::endl;
-        // ModelPart& CurrentModelPart = this->GetModelPart();
-        // this->Clear();
-
-        // Reset displacements to the initial (Assumes Water Pressure is the convergence criteria)
-        /* block_for_each(CurrentModelPart.Nodes(), [&](Node& rNode) {
-             auto dold = rNode.GetSolutionStepValue(WATER_PRESSURE, 1);
-             rNode.GetSolutionStepValue(WATER_PRESSURE, 0) = dold;
-             });*/
 
         GeoMechanicsNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>::InitializeSolutionStep();
         GeoMechanicsNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>::Predict();
         return GeoMechanicsNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>::SolveSolutionStep();
     }
 
+    virtual void BaseClassFinalizeSolutionStep()
```

Comment:
Just a remark. My understanding is that this member function has been introduced just to avoid actually solving the piping problem in our unit tests. In general, I consider it poor practice to add API to a class for testing purposes only. In this case, I can accept it (at least temporarily), since all we want to do is to have some unit tests in place for an _existing_ class. I can imagine that as we move forward and we start to refactor this class, at some point we may want to remove this member again.